File: mobile/app/src/main/java/com/flightrisk/app/drone/TelloState.kt
package com.flightrisk.app.drone

enum class TelloConnectionState {
    DISCONNECTED,
    CONNECTING,
    CONNECTED,
    STREAMING,
    ERROR,
}

data class TelloTelemetry(
    val battery: Int = 0,
    val height: Int = 0,
    val temperature: Int = 0,
    val flightTime: Int = 0,
    val isFlying: Boolean = false,
)

data class TelloState(
    val connectionState: TelloConnectionState = TelloConnectionState.DISCONNECTED,
    val telemetry: TelloTelemetry = TelloTelemetry(),
    val errorMessage: String? = null,
    val isOnTelloWifi: Boolean = false,
)

File: ios/FlightRisk/Drone/TelloState.swift
import Foundation

/// Connection lifecycle states for the Tello UDP link.
enum TelloConnectionState: String {
    case disconnected, connecting, connected, streaming, error
}

/// Telemetry snapshot from periodic state polling.
struct TelloTelemetry {
    var battery: Int? = nil    // nil = no reading yet, 0 = genuinely 0%
    var height: Int = 0
    var temperature: Int = 0
    var flightTime: Int = 0
    var isFlying: Bool = false
}

/// Observable aggregate of connection + telemetry + error state.
struct TelloState {
    var connectionState: TelloConnectionState = .disconnected
    var telemetry: TelloTelemetry = TelloTelemetry()
    var errorMessage: String? = nil
    var isOnTelloWifi: Bool = false
}
